### PR TITLE
Refactor ixblue_c3_ins package

### PR DESCRIPTION
**Pull request by hidmic**
# Description

This patch refactors the `ixblue_c3_ins` package to:

- Publish an auv_interfaces::State message (as introduced in #86)
- Gracefully terminate on error and on Ctrl+C.
- Normalize style and layout, and setup linters.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [ ] Unit tests are passing
- [x] Linter tests are passing

# How To Test

## Test A

Launch INS driver node.

```sh
roslaunch ixblue_c3_ins ixblue_c3_ins.launch
```

Use `rostopic echo` to make sure it is publishing as expected.

## Test B

Verify `c3_ins.py` emulator-based demo is working (as described in the README.txt)

### CODE REVIEW

**Comment by hidmic** [comment]
I have yet to test this on the vehicle.

File: ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public:
+  ixBlueC3InsDriver();
+  ~ixBlueC3InsDriver();
+
+  void spin();
+
+private:
+  void publish(const c3_protocol::nav_long::nav_long_data_t& data);
+
+  ros::NodeHandle nh_;
```
Comment (patapufate) [inline ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h:34]: Why do we have 2 nodeHandler?

File: ixblue_c3_ins/CMakeLists.txt [hunk 2]
```diff
@@ -1,99 +1,36 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(ixblue_c3_ins)
 
-set(NODE_APP_NAME ${PROJECT_NAME}_node)
-
 ## Compile as C++11, supported in ROS Kinetic and newer
 add_compile_options(-std=c++11)
 
 ## Find catkin macros and libraries
```
Comment (patapufate) [inline ixblue_c3_ins/CMakeLists.txt:7]: nit. remove unnecessary boilerplate

File: ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ */
+class nav_long
+{
+public:
+  // constructor/destructor
+  nav_long(char *buf, ssize_t nbytes);
+  ~nav_long();
+  // accessors
+
+  struct nav_long_data_t
```
Comment (patapufate) [inline ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h:34]: I see different data types between nav_long_data_t struct and NavigationLong.msg eg:
- int32_t latitude / float64 latitude

**Review by patapufate — CHANGES_REQUESTED** [review]

File: ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h (hunk 1 above)
Comment (hidmic) [reply ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h:34]: A public node handle and a private node handle. The public one is at the root namespace, and it is useful for topic interfaces ('/state'). The private one uses the fully qualified node name as namespace, and it is useful for parameters (`/ins_name/some_param`).

File: ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h (hunk 3 above)
Comment (hidmic) [reply ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h:34]: That's right ! The INS sends latitude and longitude information as signed 32 bit integers with a quantization constant (check NAVIGATION LONG packet in the [INS manual](https://www.igp.de/manuals/7-INS-InterfaceLibrary_MU-INSIII-AN-001-O.pdf)), but we publish them as floating point quantities after applying the corresponding constant.

File: ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h (hunk 1 above)
Comment (patapufate) [reply ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h:34]: Ok

File: ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h (hunk 3 above)
Comment (patapufate) [reply ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h:34]: Ok

**Comment by hidmic** [comment]
Rebased on top #93. Added missing diagnostic checks. We'll also need to figure out how to deal with this changes in the ROS graph at the `health_monitor` level.

**Comment by hidmic** [comment]
Rebased after merging #93.

**Review by patapufate — APPROVED** [review]
LGTM!

**Comment by hidmic** [comment]
I've also tested this on the vehicle. Moving forward !

**Pull request merged**